### DISCUSSION TITLE
Developer Portal Reports Page - Add Date Input for Filtering

### DISCUSSION THREAD

**Discussion by kevinpearce** [Azure/api-management-developer-portal#2629] [Feature request]
## Feature Request

End Users that adhere to the [Microsoft Quota Policy](https://learn.microsoft.com/en-us/azure/api-management/api-management-sample-flexible-throttling#quotas) have no way to check their monthly stats on the Reports page when Subscriptions start part way though a month.
I believe this feature was present on the Deprecated Developer Portal and was not replicated.

Please add a way for End Users to specify both a Start and End date on the Reports page.

## Potential Reproduction steps

[Paperbits](https://paperbits.io/) already offers some form of Form/Date input, so it should be feasible that this could be implemented considering this underpins most, if not all of the Developer Portal.

https://github.com/paperbits/paperbits-forms/tree/master/src/date-input

Suggestion - add a date picker option in the empty space next to existing buttons?

![apim_report_suggestion](https://github.com/Azure/api-management-developer-portal/assets/42847716/802392e1-61e0-43d7-be0e-4abe589efbaf)

The existing buttons appear to just be setting background dates to populate various REST calls so these could easily take the date input without much effort in their StartTime/EndTime params.

https://github.com/Azure/api-management-developer-portal/blob/master/src/components/reports/ko/runtime/reports.html#L2-L11
https://github.com/Azure/api-management-developer-portal/blob/master/src/components/reports/ko/runtime/reports.ts#L531-L558

## Expected behavior

End Users should have a way to check monthly quota's and this should be considered a CORE part of APIM functionality.

## Is your portal managed or self-hosted?

N/A

## Release tag or commit SHA (if using self-hosted version)

N/A

## API Management service name

N/A

## Environment

N/A

## Additional context

Also relevant (but very old and way down the backlog) - Issue #733 

I would have used the https://aka.ms/apimwish link as specified above but checking it - it requires some heavy SPAM Bot filtering as it is unfortunately currently useless.

**Comment by sthirthala** [comment]
Thanks for the input. We will add this to our feature request pipeline
